app/src/org/commcare/activities/FormRecordListActivity.java

```diff
@@ -464,10 +464,13 @@ private void createFormRecordScanResultDialog(Pair<Boolean, String> result, Form
                 result.second, resId, null);
 
         if (FormRecordFilter.Pending.containsStatus(record.getStatus())) {
-            dialog.setNegativeButton(Localization.get("app.workflow.forms.quarantine"), (dialog1, which) -> {
-                manuallyQuarantineRecord(record);
-                dismissAlertDialog();
-            });
+            if (DeveloperPreferences.isManualFormQuarantineAllowed()) {
+                dialog.setNegativeButton(Localization.get("app.workflow.forms.quarantine"), (dialog1, which) -> {
+                    manuallyQuarantineRecord(record);
+                    dismissAlertDialog();
+                    DeveloperPreferences.disableManualFormQuarantine();
```

Comment:
Is there a reason this is "spring loaded" so to speak and only lets you quarantine a single form? That will get quite painful if there is some strange event that causes issues for a lot of forms in a row.